packages/core/src/agent/AgentConfig.ts

```diff
@@ -11,12 +11,14 @@ import { DidCommMimeType } from '../types'
 
 export class AgentConfig {
   private initConfig: InitConfig
+  private initConfigEndpoints: string[]
   public label: string
   public logger: Logger
   public readonly agentDependencies: AgentDependencies
 
   public constructor(initConfig: InitConfig, agentDependencies: AgentDependencies) {
     this.initConfig = initConfig
+    this.initConfigEndpoints = initConfig.endpoints ?? [DID_COMM_TRANSPORT_QUEUE]
```

Comment:
This implementation changes the behaviour, and I think we should be careful here.

Previously if you would pass an empty `endpoints` array it would still return the `DID_COMM_TRANSPORT_QUEU` endpoint. Now it will use an empty array.

If I also later set it as an empty array it will actually use that empty array, and not defualt back to the `DID_COMM_TRANSPORT_QUEU` variable. 

Maybe we can structure it like this:
- We add a new, optional, private value  _endpoints to the class
- In the constructor we assign the value of `initConfig.endpoints` to the `_endpoints` variable
- We add a `set endpoints` method that assigns the input value to `_endpoints`
- We update the `get endpoints` method that will return the value of `_endpoints` if `_endpoints` is not undefined and has at least one entry. Otherwise we return `[DID_COMM_TRANSPORT_QUEUE]` as we did before. This was we don't introduce any breaking changes. This is important, because the framework assumes there's always at least one endpoint returned (hence the type `[string, ...string[]]` instead of `string[]`).

Reply:
Makes sense! On it!